File: DotNet/VB.NET/VS2013/ImageReduce/OptionsForm.vb
'*********************************************************************'
'* Copyright 2005-2016 - Accusoft Corporation, Tampa Florida.        *'
'* This sample code is provided to Accusoft licensees "as is"        *'
'* with no restrictions on use or modification. No warranty for      *'
'* use of this sample code is provided by Accusoft.                  *'
'*********************************************************************'
Imports System.Windows.Forms
Imports Accusoft.ImagXpressSdk

Partial Public Class OptionsForm
    Inherits Form
    Private _jpegQualityFactor As System.Int32 = 22
    Public Property JpegQualityFactor() As System.Int32
        Get
            Return _jpegQualityFactor
        End Get
        Set(value As System.Int32)
            _jpegQualityFactor = value
        End Set
    End Property

    Private _transformMode As TransformMode = Accusoft.ImagXpressSdk.TransformMode.None
    Public Property TransformMode() As TransformMode
        Get
            Return _transformMode
        End Get
        Set(value As TransformMode)
            _transformMode = value
        End Set
    End Property

    Public Sub New()
        InitializeComponent()
    End Sub

    Private Sub buttonCancel_Click(sender As Object, e As System.EventArgs)
        Me.DialogResult = DialogResult.Cancel
        Me.Close()
    End Sub

    Private Sub buttonOK_Click(sender As Object, e As System.EventArgs) Handles buttonOK.Click
        _jpegQualityFactor = CInt(Math.Truncate(jpegQualityFactorNumericUpDown.Value))
        _transformMode = CType(TransformModeComboBox.SelectedIndex, TransformMode)
        Me.DialogResult = DialogResult.OK
        Me.Close()
    End Sub

    Private Sub OptionsForm_Load(sender As Object, e As System.EventArgs) Handles MyBase.Load
        jpegQualityFactorNumericUpDown.Value = _jpegQualityFactor
        TransformModeComboBox.SelectedIndex = _transformMode
    End Sub
End Class

File: DotNet/C#/VS2012/ImageReduce/OptionsForm.cs
//*********************************************************************'
//* Copyright 2005-2016 - Accusoft Corporation, Tampa Florida.        *'
//* This sample code is provided to Accusoft licensees "as is"        *'
//* with no restrictions on use or modification. No warranty for      *'
//* use of this sample code is provided by Accusoft.                  *'
//*********************************************************************'
using System.Windows.Forms;
using Accusoft.ImagXpressSdk;

namespace ImageReduce
{
    public partial class OptionsForm : Form
    {
        private TransformMode _tranformMode = TransformMode.None;
        public TransformMode TransformMode
        {
            get
            {
                return _tranformMode;
            }
            set
            {
                _tranformMode = value;
            }
        }

        private System.Int32 _jpegQualityFactor = 22;
        public System.Int32 JpegQualityFactor
        {
            get
            { 
                return _jpegQualityFactor; 
            }
            set
            {   
                _jpegQualityFactor = value; 
            }
        }

        public OptionsForm()
        {
            InitializeComponent();
        }

        private void buttonCancel_Click(object sender, System.EventArgs e)
        {
            this.DialogResult = DialogResult.Cancel;
            this.Close();
        }

        private void buttonOK_Click(object sender, System.EventArgs e)
        {
            _jpegQualityFactor = (int)jpegQualityFactorNumericUpDown.Value;
            _tranformMode = (TransformMode)TransformModeComboBox.SelectedIndex;
            this.DialogResult = DialogResult.OK;
            this.Close();
        }

        private void OptionsForm_Load(object sender, System.EventArgs e)
        {
            jpegQualityFactorNumericUpDown.Value = _jpegQualityFactor;
            TransformModeComboBox.SelectedIndex = (int)_tranformMode;
        }
    }
}
